wgpu-core/src/command/render.rs

```diff
@@ -1103,10 +1103,26 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                     // Rebind resource
                     if state.binder.pipeline_layout_id != Some(pipeline.layout_id.value) {
                         let pipeline_layout = &pipeline_layout_guard[pipeline.layout_id.value];
-                        state.binder.pipeline_layout_id = Some(pipeline.layout_id.value);
-                        state
+
+                        let old_pipeline_layout_id = state
                             .binder
-                            .reset_expectations(pipeline_layout.bind_group_layout_ids.len());
+                            .pipeline_layout_id
+                            .replace(pipeline.layout_id.value);
+                        let reset_start = if let Some(old_id) = old_pipeline_layout_id {
```

Comment:
can we move that logic into `binder` perhaps to not copy between the compute/render?